Frontend/app/src/main/java/io/happybugs/happybugs/ReportActivity.java

```diff
@@ -0,0 +1,143 @@
+                (View) findViewById(R.id.whenView), (View) findViewById(R.id.whoView),
+                (View) findViewById(R.id.detailsView));
+
+        buttons.whatBtn.setOnClickListener(this);
+        buttons.whereBtn.setOnClickListener(this);
+        buttons.whenBtn.setOnClickListener(this);
+        buttons.whoBtn.setOnClickListener(this);
+        buttons.detailsBtn.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()){
+            case R.id.whatBtn:
+                enableTextView(editTexts.whatText, editTexts.whereText,
+                        editTexts.whenText, editTexts.whoText, editTexts.detailsText);
+                enableBoxes(editTexts.whatText, checkBoxes.whatCheck, buttons.saveBtn);
+                enableView(editTexts.whatText, views.whatView,
+                        views.whereView, views.whenView, views.whoView, views.detailsView);
+                break;
+            case R.id.whereBtn:
+                enableTextView(editTexts.whereText, editTexts.whatText,
+                        editTexts.whenText, editTexts.whoText, editTexts.detailsText);
+                enableBoxes(editTexts.whereText, checkBoxes.whereCheck, buttons.saveBtn);
+                enableView(editTexts.whereText, views.whereView,
+                        views.whatView, views.whenView, views.whoView, views.detailsView);
+                break;
+            case R.id.whenBtn:
+                enableTextView(editTexts.whenText, editTexts.whatText,
+                        editTexts.whereText, editTexts.whoText, editTexts.detailsText);
+                enableBoxes(editTexts.whenText, checkBoxes.whenCheck, buttons.saveBtn);
+                enableView(editTexts.whenText, views.whenView,
+                        views.whatView, views.whereView, views.whoView, views.detailsView);
+                break;
+            case R.id.whoBtn:
+                enableTextView(editTexts.whoText, editTexts.whatText,
+                        editTexts.whereText, editTexts.whenText, editTexts.detailsText);
+                enableBoxes(editTexts.whoText, checkBoxes.whoCheck,buttons.saveBtn);
+                enableView(editTexts.whoText, views.whoView,
+                        views.whatView, views.whereView, views.whenView, views.detailsView);
+                break;
+            case R.id.detailsBtn:
+                enableTextView(editTexts.detailsText, editTexts.whatText,
+                        editTexts.whereText, editTexts.whenText, editTexts.whoText);
+                enableBoxes(editTexts.detailsText, checkBoxes.detailsCheck, buttons.saveBtn);
+                enableView(editTexts.detailsText, views.detailsView,
+                        views.whatView, views.whereView, views.whenView, views.whoView);
+                break;
+        }
+    }
+
+    //Open and collapse edit text on button click
+    protected void enableTextView(TextView tv1, TextView tv2,
```

Comment:
I think this function needs an explanation (or a better name). The name suggests it's enabling a text view but all parameters are TextViews. Also, if we decide to add another text view in the report page, will this be able to handle that?

Reply:
I found the variables and functions lack explanation. added explanation and also renamed the functions. 
I also figured it would be hard to handle if more text views are added. After some thoughts, made functions for each ReportEditTexts and ReportViews classes. This reduced the number of  parameters needed for "enable~" functions. In addition it made possible to create a new function to reduce the repeated blocks that you pointed out. Still think it would be struggling when more views are added on the page, but for now the functions are refactored and seems to have improved. Thank you!

Reply:
If the purpose of enableTextView is to enable one text view and disable all others, can we just pass in the view that needs to be enabled? And within this function, all other text views can be disabled. I think this will reduce the number of parameters. Do you think this approach will work here?